submissions/north_sea_mountain.jl

```diff
@@ -0,0 +1,19 @@
+author = "Tim Reichelt"
+description = "North Sea mountain"
+
+using SpeedyWeather, RainMaker
+
+spectral_grid = SpectralGrid(trunc=31, nlayers=8)
+model = PrimitiveWetModel(; spectral_grid)
+model.feedback.verbose = false # hide
+
+rain_gauge = RainGauge(spectral_grid, lond=-1.25, latd=51.75)
+add!(model, rain_gauge)
+
+simulation = initialize!(model)
+
+# add a massive mountain at 51.75°N, 0°W, *after* model initialization
+H, λ₀, φ₀, σ = 4000, 2, 51, 5     # height, lon, lat position, and width
+set!(model, orography=(λ,φ) -> H*exp((-(λ-λ₀)^2 - (φ-φ₀)^2)/2σ^2), add=true)
```

Comment:
I've been wondering whether such a mountain crossing the prime meridian would actually be constructed correctly given the jump in longitudes from <360 back to 0 ... I'm wondering how to best fix that.

Reply:
I think the problem here stems from the fact that the function `(λ,φ) -> H*(exp((-(λ-λ₀)^2 - (φ-φ₀)^2)/2σ^2) + exp((-(λ-(λ₀+360))^2 - (φ-φ₀)^2)/2σ^2))` is using a Euclidean distance function when the underlying geometry of the space is actually non-Euclidean. So instead calculating the distances between (lon, lat) pairs using geodesics should fix this. 

I have tried using the [Geodesics.jl](https://github.com/anowacki/Geodesics.jl) package but the following code takes prohibitively long (doesn't terminate within 10 minutes):
```julia
earth_geodesic((lon1, lat1), (lon2, lat2)) = Geodesics.inverse(deg2rad.((lon1, lat1, lon2, lat2))..., Geodesics.EARTH_R_MAJOR_WGS84, Geodesics.F_WGS84)[1]
set!(model, orography=(λ,φ) -> H*exp((- earth_geodesic((λ,φ), (λ₀, φ₀))^2)/2σ^2), add=true)
```
It seems to mostly be stuck in this loop when calculating the geodesic (https://github.com/anowacki/Geodesics.jl/blob/84ba6b976bfac2edbc65c9be5189ebef53b6b99e/src/Geodesics.jl#L203). This might be due to the failure of Vincenty's algorithm to converge for [nearly antipodal points](https://en.wikipedia.org/wiki/Vincenty's_formulae#Nearly_antipodal_points).